### PR TITLE
chore: Temporarily disable the license report generation to speed up build

### PR DESCRIPTION
**Pull request by Roiocam**
# Motivation

Refs: original from #326 and #332, then #1019 

This PR improves the speed of paradox build  by disabling `sbt-license-report` on specific build.

PTAL, @mdedetrich @He-Pin

### CODE REVIEW

File: .github/workflows/link-validator.yml
Comment (Roiocam) [inline .github/workflows/link-validator.yml:38]: skip the `license-report` page generation in link-validator step.

File: project/ProjectIndexGenerator.scala
Comment (Roiocam) [inline project/ProjectIndexGenerator.scala:55]: Use the CLI option to selectively generate the index.

File: scripts/link-validator.conf
Comment (Roiocam) [inline scripts/link-validator.conf:83]: Does not need those link whitelist cause we can skip the license report now.

**Review by Roiocam — COMMENTED** [review]
Approach that skip license-report

**Comment by Roiocam** [comment]
Looks like last CI has passed. And since the PR was opened, every `Link Validator` job in each commit has been successfully passed. It seems that this approach can address the issues caused by the license report.

**Comment by mdedetrich** [comment]
I had another way of solving this issue which involved adding an `ignore-files` config to my forked https://github.com/mdedetrich/site-link-validator (the fork is also there so we don't rely on akka). Such a configuration would provide a far simpler solution as you would just do something like

```
ignore-files = "link-validator.html"
```

in `scripts/link-validator.conf`

Which makes this really annoying because a lot of time was spent on this :/

**Comment by Roiocam** [comment]
> I had another way of solving this issue which involved adding an `ignore-files` config to my forked https://github.com/mdedetrich/site-link-validator (the fork is also there so we don't rely on akka). 

I have to say that this approach also crossed my mind initially. However, considering the additional maintenance cost of another repo, I have decided to abandon this solution for this issue and consider a simpler approach.

This PR approach more than ignores license report files and could skip the task of license report generation, it can make PR checks run faster.

In addition, maintaining a new repo could provide ongoing solutions for future issues that may arise. However, the biggest obstacle for me is the sustainability of maintaining an additional repo. Throughout my career, I have encountered situations where repositories were no longer updated due to a lack of maintenance personnel, such as Apache Common DButils.

 Anyway, I am willing to accept whichever solution is ultimately accepted because both of them address the problem.

Best regards.

**Comment by mdedetrich** [comment]
So my biggest apprehension is the complexity of the build which this PR does increase, I have an overarching goal in my head to reduce to try and gradually simplify the sbt build (I haven't really formalized this yet) as I see that sbt's build complexity is creating a barrier of entry to those trying to contribute.

The reduction of the build time I don't see as a critical priority in this specific case, also do note that there will be movement in the future to try and procure our own hardware.

**Comment by Roiocam** [comment]
@mdedetrich Another view is that this undertaking is of modest scale and does not necessitate regular updates. Consequently, the absence of maintenance is not a significant concern to be pondered upon.

Scala is not my primary language, but I still can make the commit like this PR. I acknowledge that the sbt build can be a bit complex, but also offers a lot of possibilities. Considering that sbt builds are not something that needs frequent changes, I believe it won't hinder those who are trying to contribute.

Perhaps we can merge this PR first to avoid those annoying check failure. Once the PR for the other repo is merged and released, we can remove the redundant build task from this PR (This PR also includes a fix part for the broken links). Unless the repo of https://github.com/mdedetrich/site-link-validator was proceeding well.

**Comment by Roiocam** [comment]
@mdedetrich any feedback of this issue will be great.

File: .github/workflows/link-validator.yml
Comment (He-Pin) [reply .github/workflows/link-validator.yml:38]: As this is a ci, can we keep it? or we add a dedicated task for `genlicensereport`

File: project/ProjectIndexGenerator.scala
Comment (He-Pin) [inline project/ProjectIndexGenerator.scala:54]: I think we can add a `paradoxFast` and `paradoxFastBrowse` task which turn this off?

File: docs/src/main/paradox/project/project-index.md
Comment (He-Pin) [inline docs/src/main/paradox/project/project-index.md:21]: seems this can be split to another pr.

**Comment by mdedetrich** [comment]
So I am not sure how this PR relates to https://github.com/apache/incubator-pekko/issues/1019 specifically but as explained [here](https://github.com/apache/incubator-pekko/issues/1019#issuecomment-1903371167) the majority of time is taken up in sbt-license-report's `updateLicenses` task.

I would divert effort into solving that fundamental issue

File: project/ProjectIndexGenerator.scala
Comment (Roiocam) [reply project/ProjectIndexGenerator.scala:54]: Have tried those approach:

- `(Compile / paradox / javaOptions += "-Dpekko.genlicensereport.enabled=false")`
-  `(javaOptions += "-Dpekko.genlicensereport.enabled=false")`
- `Compile / paradoxMarkdownToHtml / sourceGenerators := {
       (Compile / paradoxMarkdownToHtml / sourceGenerators).value.filterNot(_ == Paradox.licenseReportGeneratorTask.taskValue)
    }`
- `Compile / paradoxMarkdownToHtml / sourceGenerators := {
      Seq.empty
    }`

None of them can be filtered in the license report generated, will keep find a new approach tomorrow.

File: docs/src/main/paradox/project/project-index.md
Comment (Roiocam) [reply docs/src/main/paradox/project/project-index.md]: I have rolled back these lines, and to avoid any misunderstanding I provide some redundant explanations here:

- The purpose of changing these lines was to allow ProjectIndexGenerator to match the text and skip writing to this file (when license report is enabled). After I complete the rollback commit, this file will always perform write operations.
- ProjectIndexGenerator cannot be completed in a separate PR because when LicenseReport is disabled, the link in the index file will complain lack of a file, resulting in a compilation failure.

**Comment by pjfanning** [comment]
@Roiocam could I ask you to get an iCLA? https://www.apache.org/licenses/contributor-agreements.html

It makes the team's life easier if regular contributors have iCLAs

**Comment by Roiocam** [comment]
> @Roiocam could I ask you to get an iCLA? https://www.apache.org/licenses/contributor-agreements.html
> 
> It makes the team's life easier if regular contributors have iCLAs

I was sent the ICLA submission on 2023/9/23, is any thing I have missed?

**Comment by mdedetrich** [comment]
So I am fine with this being a temporary fix but when https://github.com/sbt/sbt-license-report/issues/87 is resolved I would suggest to revert this commit because there isn't any reason to have this feature disable i.e. its currently a workaround for that bug.

This is because when https://github.com/sbt/sbt-license-report/issues/87 is resolved, the `updateLicenses` task should be near instant because it will actually re-use all of the dependency resolution that was already done when sbt is loaded.

**Comment by Roiocam** [comment]
> So I am fine with this being a temporary fix but when [sbt/sbt-license-report#87](https://github.com/sbt/sbt-license-report/issues/87) is resolved I would suggest to revert this commit because there isn't any reason to have this feature disable i.e. its currently a workaround for that bug.

Yes, that's why we still need another ISSUE #1019  to associate the problem.

> This is because when [sbt/sbt-license-report#87](https://github.com/sbt/sbt-license-report/issues/87) is resolved, the `updateLicenses` task should be near instant because it will actually re-use all of the dependency resolution that was already done when sbt is loaded.

No doubt it, this PR is just sugar for contributors now.

**Review by mdedetrich — APPROVED** [review]
Okay I am going to approve it, would be good if @pjfanning and @He-Pin also looked at it. @Roiocam Can you make an standard issue to revert this PR once https://github.com/sbt/sbt-license-report/issues/87 is solved so we don't forget.

Thanks for the work!

**Comment by Roiocam** [comment]
> Okay I am going to approve it, would be good if @pjfanning and @He-Pin also looked at it. @Roiocam Can you make an standard issue to revert this PR once [sbt/sbt-license-report#87](https://github.com/sbt/sbt-license-report/issues/87) is solved so we don't forget.

Yes, I will stay updated on the progress.

**Review by He-Pin — APPROVED** [review]
Thanks

**Comment by He-Pin** [comment]
@pjfanning Would you like to take a look at this, I think with this , all our contributors can have a faster local development exprirence.

**Pull request merged**
